File: Src/java/engine/src/main/java/org/opencds/cqf/cql/engine/elm/execution/ToListEvaluator.java
package org.opencds.cqf.cql.engine.elm.execution;

import java.util.Collections;

import org.opencds.cqf.cql.engine.execution.Context;

public class ToListEvaluator extends org.cqframework.cql.elm.execution.ToList {

    public static Object toList(Object operand)
    {
        // check to see if it is already a list
        if (operand instanceof Iterable)
        {
            return operand;
        }

        return operand == null ? Collections.emptyList() : Collections.singletonList(operand);
    }

    @Override
    protected Object internalEvaluate(Context context)
    {
        return toList(getOperand().evaluate(context));
    }
}

File: Src/java/engine/src/commonMain/kotlin/org/opencds/cqf/cql/engine/elm/executing/ToListEvaluator.kt
package org.opencds.cqf.cql.engine.elm.executing

import kotlin.jvm.JvmStatic
import org.opencds.cqf.cql.engine.runtime.List
import org.opencds.cqf.cql.engine.runtime.Value
import org.opencds.cqf.cql.engine.runtime.toCqlList

object ToListEvaluator {
    @JvmStatic
    fun toList(operand: Value?): List {
        // check to see if it is already a list
        if (operand is List) {
            return operand
        }

        return (if (operand == null) mutableListOf() else mutableListOf(operand)).toCqlList()
    }
}
